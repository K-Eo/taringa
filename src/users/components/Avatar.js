import React, { PureComponent } from 'react'
import { connect } from 'react-redux'

export class Avatar extends PureComponent {
  render() {
    const { firstname, lastname, avatar } = this.props
    const fullname = `${firstname} ${lastname}`

    return (
      <img
        alt={fullname}
        className="rounded Commentable-avatar"
        src={avatar}
        title={fullname}
      />
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return state.entities.users[ownProps.id]
}

export default connect(mapStateToProps)(Avatar)
