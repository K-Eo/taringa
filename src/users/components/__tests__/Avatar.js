import React from 'react'

import { Avatar } from '../Avatar'

describe('Avatar', () => {
  it('exists', () => {
    expect(Avatar).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Avatar />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with props', () => {
    const wrapper = shallow(
      <Avatar avatar="baz" firstname="foo" lastname="bar" />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
