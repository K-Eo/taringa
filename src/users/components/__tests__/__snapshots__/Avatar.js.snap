// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders default 1`] = `
<img
  alt="#FIRSTNAME #LASTNAME"
  className="rounded Commentable-avatar"
  src="#AVATAR"
  title="#FIRSTNAME #LASTNAME"
/>
`;

exports[`Avatar renders with props 1`] = `
<img
  alt="foo bar"
  className="rounded Commentable-avatar"
  src="baz"
  title="foo bar"
/>
`;
